medmodels/treatment_effect_estimation/analysis_modules/matching/matching.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING, Literal, Set, Tuple
+
+import polars as pl
+
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.types import MedRecordAttributeInputList, NodeIndex
+
+if TYPE_CHECKING:
+    import sys
+
+    if sys.version_info >= (3, 10):
+        from typing import TypeAlias
+    else:
+        from typing_extensions import TypeAlias
+
+MatchingMethod: TypeAlias = Literal["propensity", "nearest_neighbors"]
+
+
+class Matching(metaclass=ABCMeta):
+    """
+    The Base Class for matching.
+    """
+
+    def _preprocess_data(
```

Comment:
this whole function should not just take all of the data out of the medrecord into a polars DataFrame to do some calculations.

Please create a issue describing the functionality needed for the MedRecord to support @MarIniOnz @LauraBoenchenLB 

For now it is fine though